lms/djangoapps/grades/signals/handlers.py

```diff
@@ -70,51 +65,30 @@ def submissions_score_reset_handler(sender, **kwargs):  # pylint: disable=unused
     course_id = kwargs['course_id']
     usage_id = kwargs['item_id']
     user = user_by_anonymous_id(kwargs['anonymous_user_id'])
+    if user is None:
+        return
 
     SCORE_CHANGED.send(
         sender=None,
         points_possible=0,
         points_earned=0,
-        user=user,
+        user_id=user.id,
         course_id=course_id,
         usage_id=usage_id
     )
 
 
 @receiver(SCORE_CHANGED)
-def recalculate_subsection_grade_handler(sender, **kwargs):  # pylint: disable=unused-argument
+def enqueue_subsection_update(sender, **kwargs):  # pylint: disable=unused-argument
     """
-    Consume the SCORE_CHANGED signal and trigger an update.
-    This method expects that the kwargs dictionary will contain the following
-    entries (See the definition of SCORE_CHANGED):
-       - points_possible: Maximum score available for the exercise
-       - points_earned: Score obtained by the user
-       - user: User object
-       - course_id: Unicode string representing the course
-       - usage_id: Unicode string indicating the courseware instance
+    Handles the SCORE_CHANGED signal by enqueueing a subsection update to occur asynchronously.
     """
-    student = kwargs['user']
-    course_key = CourseLocator.from_string(kwargs['course_id'])
-    if not PersistentGradesEnabledFlag.feature_enabled(course_key):
-        return
-
-    scored_block_usage_key = UsageKey.from_string(kwargs['usage_id']).replace(course_key=course_key)
-    collected_block_structure = get_course_in_cache(course_key)
-    course = get_course_by_id(course_key, depth=0)
-
-    subsections_to_update = collected_block_structure.get_transformer_block_field(
-        scored_block_usage_key,
-        GradesTransformer,
-        'subsections',
-        set()
-    )
-    subsection_grade_factory = SubsectionGradeFactory(student, course, collected_block_structure)
-    for subsection_usage_key in subsections_to_update:
-        transformed_subsection_structure = get_course_blocks(
-            student,
-            subsection_usage_key,
-            collected_block_structure=collected_block_structure,
-        )
-        subsection_grade_factory.update(
-            transformed_subsection_structure[subsection_usage_key], transformed_subsection_structure
+    kwargs.pop('signal', None)  # This value is not needed by the task and cannot be serialized properly
+    log.info(
```

Comment:
I'd bump it down to debug. If there's a problem, we will be glad to have it on a sandbox, right?